base/applications/mspaint/canvas.cpp

```diff
@@ -13,20 +13,79 @@ CCanvasWindow canvasWindow;
 /* FUNCTIONS ********************************************************/
 
 CCanvasWindow::CCanvasWindow()
-    : m_whereHit(HIT_NONE)
-    , m_ptOrig { 0, 0 }
+    : m_drawing(FALSE)
+    , m_hitSelection(HIT_NONE)
+    , m_whereHit(HIT_NONE)
+    , m_ptOrig { -1, -1 }
 {
+    ::SetRectEmpty(&m_rcNew);
+}
+
+VOID CCanvasWindow::drawZoomFrame(INT mouseX, INT mouseY)
+{
+    // FIXME
```

Comment:
What specifically does not work here yet?

Reply:
Added comment.